src/MatchInterface.php

```diff
@@ -15,13 +15,13 @@ interface MatchInterface
      *
      * @return string
      */
-    public function getName();
+    public function getName(): string;
 
     /**
      * Matches the pattern against data.
      *
      * @param  string $pattern
      * @return mixed
      */
-    public function matches($pattern);
+    public function matches(string $pattern);
```

Comment:
Please do not add native php typehints in this PR as this introduces a BC break. 
Focus on PHP 8.0 compatibility only.
Thanks!